### PR TITLE
Integrated login module

### PR DESCRIPTION
**Pull request by RajuKoushik**
Signed-off-by: Raju Koushik g.rajukoushik@gmail.com

### CODE REVIEW

**Review by singh1114 — CHANGES_REQUESTED** [review]
Not acceptable. Add at least something into the files. Put css in the login.html to the css file and javascript to login.js file.

File: scanapp/templates/scanapp/login.html [hunk 1]
```diff
@@ -1,156 +1,26 @@
-{% extends 'scanapp/base.html' %}
 {% load staticfiles %}
 <html>
```
Comment (singh1114) [inline scanapp/templates/scanapp/login.html]: No need for this.

File: scanapp/templates/scanapp/login.html [hunk 2]
```diff
@@ -1,156 +1,26 @@
...
 
 
 <head>
```
Comment (singh1114) [inline scanapp/templates/scanapp/login.html]: Same here. Check how is it done in index.html

File: scanapp/templates/scanapp/login.html [hunk 3]
```diff
@@ -1,156 +1,26 @@
...
-        .btn-register:focus {
-            color: #fff;
-            background-color: #1CA347;
-            border-color: #1CA347;
-        }
-
-    </style>
-
-
 </head>
```
Comment (singh1114) [inline scanapp/templates/scanapp/login.html]: This is in the base.html.

File: scanapp/templates/scanapp/login.html [hunk 4]
```diff
@@ -254,26 +124,7 @@
...
-            $("#register-form").delay(100).fadeIn(100);
-            $("#login-form").fadeOut(100);
-            $('#login-form-link').removeClass('active');
-            $(this).addClass('active');
-            e.preventDefault();
-        });
-
-    });
-</script>
+<script src="{% static "javascript/login.js" %}"></script>
```
Comment (singh1114) [inline scanapp/templates/scanapp/login.html]: scripts don't get loaded this way.

**Review by singh1114 — CHANGES_REQUESTED** [review]

File: scanapp/templates/scanapp/login.html (hunk 4 above)
Comment (RajuKoushik) [reply scanapp/templates/scanapp/login.html]: Did you run the code @singh1114 . Coz it's running fine and the script works

Comment (RajuKoushik) [reply scanapp/templates/scanapp/login.html]: Updates the changes @singh1114

Comment (singh1114) [reply scanapp/templates/scanapp/login.html]: Ok. Just wait till I remove every bad code and then I will test locally.

Comment (RajuKoushik) [reply scanapp/templates/scanapp/login.html]: cool

File: scanapp/views.py [hunk 5]
```diff
@@ -121,3 +128,29 @@ def get(self, request, *args, **kwargs):
 
 def login(request):
```
Comment (singh1114) [inline scanapp/views.py]: Use class based views.

File: scanapp/views.py [hunk 6]
```diff
@@ -121,3 +128,29 @@ def get(self, request, *args, **kwargs):
...
     return render(request, 'scanapp/login.html')
+
+
+@csrf_exempt
+def post_sign_up(request):
```
Comment (singh1114) [inline scanapp/views.py]: Use class-based views. We don't want to add csrf_exception for this view. Do we?

Comment (RajuKoushik) [reply scanapp/views.py]: I'll use the class based views but we need the csrf exemt. Just check the docs.

**Comment by RajuKoushik** [comment]
@singh1114  just updated with the class based views.

File: scancodeserver/settings.py [hunk 7]
```diff
@@ -117,9 +117,9 @@
 DATABASES = {
     'default': {
         'ENGINE': 'django.db.backends.postgresql',
-        'NAME': 'DATABASE_NAME',
-        'USER': 'DATABASE_USER',
-        'PASSWORD': 'USER_PASSWORD',
+        'NAME': 'taskbuster_db',
```
Comment (singh1114) [inline scancodeserver/settings.py]: BAD. Check before committing.

File: scanapp/views.py [hunk 8]
```diff
@@ -126,31 +127,31 @@ def get(self, request, *args, **kwargs):
...
+                password=request.POST.get('password'),
+                email=request.POST.get('email')
+            )
+            # user.save()
 
-        # user.first_name = request.POST.get('first_name', '')
-        # user.last_name = request.POST.get('last_name', '')
-        user.save()
+            # user.first_name = request.POST.get('first_name', '')
+            # user.last_name = request.POST.get('last_name', '')
```
Comment (singh1114) [inline scanapp/views.py]: Remove these extra comments.

**Review by singh1114 — CHANGES_REQUESTED** [review]

File: scancodeserver/settings.py (hunk 7 above)
Comment (RajuKoushik) [reply scancodeserver/settings.py]: Fixed it in the minor glitch commit.

File: scanapp/views.py (hunk 8 above)
Comment (RajuKoushik) [reply scanapp/views.py]: Done !

**Pull request merged**